.github/workflows/generate-doc.yml

```diff
@@ -0,0 +1,33 @@
+name: Generate documentation
+
+on:
+  push:
+    branches: ['*']
+
+jobs:
+  generate-documentation:
+    name: Generate documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2.3.4
+      - name: Install perl-doc
+        run: |
+          sudo apt-get update
+          sudo apt-get install perl perl-doc
+      - name: Run perl-doc
+        run: |
+          pwd
+          perldoc lib/ProductOpener/Food.pm
+          pod2html .lib/ProductOpener/Food.pm > Food.html
+          ls
+#ls lib/ProductOpener/
+#   - name: Convert from POD to HTML
+#     run: pod2html ../Food.pm > Food.html
+# https://metacpan.org/pod/Pod::Simple::HTMLBatch
+      - name: Deploy API documentation to Github Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: 'gh-pages' # <- Branch on which commit will be made
+          folder: .  # <- Folder containing our generated HTML docs lib/ProductOpener/
```

Comment:
Looking at the gh-pages branch, this copies the whole repo content. It could be better to output the docs to another directory.

Reply:
fixed @hangy